packages/amplify-cli/src/extensions/amplify-helpers/remove-resource.ts

```diff
@@ -97,8 +98,11 @@ export async function removeResource(
   }
 
   try {
-    return await deleteResourceFiles(context, category, resourceName, resourceDir);
+    return deleteResourceFiles(context, category, resourceName, resourceDir);
```

Comment:
This will return the promise before it is awaited, rendering the catch block useless if an exception occurs that is not synchronous. ie, only synchronous exceptions will be caught by this try/catch now

Reply:
Thanks for catching this! Fixed.